src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -34,7 +34,7 @@ internal sealed class HttpConnectionPool : IDisposable
 
         private bool _http2Enabled;
         private Http2Connection _http2Connection;
-        private SemaphoreSlim _http2ConnectionCreateLock;
+        private volatile SemaphoreSlim _http2ConnectionCreateLock;
```

Comment:
This one is also introducing a bunch of unnecessary volatile reads.

I'm not sure this SemaphoreSlim actually needs to be created lazily, but let's assume for now that it is.  We can just delete this code:
```C#
            if (_http2ConnectionCreateLock == null)
            {
                lock (SyncObj)
                {
                    if (_http2ConnectionCreateLock == null)
                    {
                        _http2ConnectionCreateLock = new SemaphoreSlim(1);
                    }
                }
            }
```
and then change the following line:
```C#
await _http2ConnectionCreateLock.WaitAsync().ConfigureAwait(false);
```
to instead be:
```C#
await LazyInitializer.EnsureInitialized(ref _http2ConnectionCreateLock, () => new SemaphoreSlim(1)).WaitAsync().ConfigureAwait(false);
```
and not add volatile to the field.
cc: @geoffkizer

Reply:
Sounds good to me.